llama-index-core/llama_index/core/llms/llm.py

```diff
@@ -396,6 +399,117 @@ class Test(BaseModel):
         result = await program.acall(**prompt_args)
         dispatcher.event(LLMStructuredPredictEndEvent(output=result))
         return result
+        
+
+    @dispatcher.span
+    def stream_structured_predict(
+        self,
+        output_cls: BaseModel,
+        prompt: PromptTemplate,
+        **prompt_args: Any,
+    ) -> Generator[Union[Model, List[Model]], None, None]:
+        r"""Stream Structured predict.
+
+        Args:
+            output_cls (BaseModel):
+                Output class to use for structured prediction.
+            prompt (PromptTemplate):
+                Prompt template to use for structured prediction.
+            prompt_args (Any):
+                Additional arguments to format the prompt with.
+
+        Returns:
+            Generator: A generator returning partial copies of the model or list of models.
+
+        Examples:
+            ```python
+            from pydantic.v1 import BaseModel
+
+            class Test(BaseModel):
+                \"\"\"My test class.\"\"\"
+                name: str
+
+            from llama_index.core.prompts import PromptTemplate
+
+            prompt = PromptTemplate("Please predict a Test with a random name related to {topic}.")
+            stream_output = llm.stream_structured_predict(Test, prompt, topic="cats")
+            for partial_output in stream_output:
+                # stream partial outputs until completion
+                print(partial_output.name)
+            ```
+        """
+        from llama_index.core.program.utils import get_program_for_llm
+
+        # dispatcher.event(
+        #     LLMStructuredPredictStartEvent(
+        #         output_cls=output_cls, template=prompt, template_args=prompt_args
+        #     )
+        # )
```

Comment:
curious why this is commented out for now?

Reply:
the end event was causing an error, because the output needs to be a BaseModel but here it's a generator 

rather than figuring it out i disabled it for now 😅